agent/fw_drupal.c

```diff
@@ -753,33 +753,21 @@ NR_PHP_WRAPPER(nr_drupal_wrap_module_invoke_all_before) {
   NR_PHP_WRAPPER_REQUIRE_FRAMEWORK(NR_FW_DRUPAL);
 
   hook_copy = nr_php_arg_get(1, NR_EXECUTE_ORIG_ARGS);
-  if (nr_php_is_zval_non_empty_string(hook_copy)) {
-    nr_stack_push(&NRPRG(drupal_module_invoke_all_hooks), hook_copy);
-    nr_stack_push(&NRPRG(drupal_module_invoke_all_states), (void*)!NULL);
-  } else {
-    nr_stack_push(&NRPRG(drupal_module_invoke_all_states), NULL);
-  }
+  invoke_all_push_hook_stacks(hook_copy);
 }
 NR_PHP_WRAPPER_END
 
-static void module_invoke_all_clean_stacks() {
-  if ((bool)nr_stack_pop(&NRPRG(drupal_module_invoke_all_states))) {
-    zval* hook_copy = nr_stack_pop(&NRPRG(drupal_module_invoke_all_hooks));
-    nr_php_arg_release(&hook_copy);
-  }
-}
-
 NR_PHP_WRAPPER(nr_drupal_wrap_module_invoke_all_after) {
   (void)wraprec;
-  module_invoke_all_clean_stacks();
+  invoke_all_clean_hook_stacks();
```

Comment:
Consider renaming this according to [this comment](https://github.com/newrelic/newrelic-php-agent/pull/701/files#r1275366315). Then it would read as:
```suggestion
  nr_drupal_invoke_all_hook_stacks_clean();
```